xwiki-commons-core/xwiki-commons-xml/src/test/java/org/xwiki/xml/XMLUtilsTest.java

```diff
@@ -279,4 +294,82 @@ void serializeNode()
         serialize = XMLUtils.serialize(node, false);
         assertEquals("<HTML><HEAD/><BODY class=\"toto\"/></HTML>", serialize);
     }
+
+    @Test
+    void extractXML() throws Exception
+    {
+        LSInput input = inputForExtractXML("<root>hello world</root>");
+        Document document = XMLUtils.parse(input);
+
+        assertEquals("<root>llo world</root>", XMLUtils.extractXML(document, 2, 40));
+    }
+
+    @Test
+    void extractXMLWithIllegalStart() throws Exception
+    {
+        LSInput input = inputForExtractXML("<root>whatever</root>");
+        Document document = XMLUtils.parse(input);
+
+        Throwable exception = assertThrows(RuntimeException.class, () -> {
+            XMLUtils.extractXML(document, -2, 40);
+        });
+        assertEquals("Failed to extract XML", exception.getMessage());
+        assertEquals(SAXException.class, exception.getCause().getClass());
+        assertEquals("Start must be greater than or equal to 0", exception.getCause().getMessage());
+    }
+
+    @Test
+    void extractXMLWithIllegalLength() throws Exception
+    {
+        LSInput input = inputForExtractXML("<root>whatever</root>");
+        Document document = XMLUtils.parse(input);
+
+        Throwable exception = assertThrows(RuntimeException.class, () -> {
+            XMLUtils.extractXML(document, 0, 0);
+        });
+        assertEquals("Failed to extract XML", exception.getMessage());
+        assertEquals(SAXException.class, exception.getCause().getClass());
+        assertEquals("Length must be greater than 0", exception.getCause().getMessage());
+    }
+
+    @Test
+    void extractXMLWithTooLongTextNode() throws Exception
+    {
+        LSInput input = inputForExtractXML("<root>hello world</root>");
+        Document document = XMLUtils.parse(input);
+
+        assertEquals("<root>llo</root>", XMLUtils.extractXML(document, 2, 4));
+        assertEquals(Level.DEBUG, logCapture.getLogEvent(0).getLevel());
+        assertEquals(String.format("Error [%s] from xml transformer", "Length limit reached"),
```

Comment:
Personally I always assert the messsge we get, I feel it's more readable than recomputing it and it allows to more easily discover formatting problems. WDYT?

Reply:
I first felt it might be more readable when separating the "inner" and "outer" message, but on second thought I agree with you.